js/coinbase.js

```diff
@@ -1633,6 +1638,165 @@ module.exports = class coinbase extends Exchange {
+         */
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'client_order_id': this.uuid (),
+            'product_id': market['id'],
+            'side': side.toUpperCase (),
+        };
+        const stopPrice = this.safeNumberN (params, [ 'stopPrice', 'stop_price', 'triggerPrice' ]);
+        const stopLossPrice = this.safeNumber (params, 'stopLossPrice');
+        const takeProfitPrice = this.safeNumber (params, 'takeProfitPrice');
+        const isStop = stopPrice !== undefined;
+        const isStopLoss = stopLossPrice !== undefined;
+        const isTakeProfit = takeProfitPrice !== undefined;
+        const timeInForce = this.safeString (params, 'timeInForce');
+        const postOnly = (timeInForce === 'PO') ? true : this.safeValue2 (params, 'postOnly', 'post_only', false);
+        let stopDirection = this.safeString (params, 'stop_direction');
+        if (type === 'limit') {
+            if (isStop) {
+                if (stopDirection === undefined) {
+                    stopDirection = (side === 'buy') ? 'STOP_DIRECTION_STOP_DOWN' : 'STOP_DIRECTION_STOP_UP';
+                }
+                request['order_configuration'] = {
+                    'stop_limit_stop_limit_gtc': {
+                        'base_size': this.amountToPrecision (symbol, amount),
+                        'limit_price': this.priceToPrecision (symbol, price),
+                        'stop_price': this.priceToPrecision (symbol, stopPrice),
+                        'stop_direction': stopDirection,
+                    },
+                };
+            } else if (isStopLoss || isTakeProfit) {
+                let triggerPrice = undefined;
+                if (isStopLoss) {
+                    if (stopDirection === undefined) {
+                        stopDirection = (side === 'buy') ? 'STOP_DIRECTION_STOP_UP' : 'STOP_DIRECTION_STOP_DOWN';
+                    }
+                    triggerPrice = this.priceToPrecision (symbol, stopLossPrice);
+                } else {
+                    if (stopDirection === undefined) {
+                        stopDirection = (side === 'buy') ? 'STOP_DIRECTION_STOP_DOWN' : 'STOP_DIRECTION_STOP_UP';
+                    }
+                    triggerPrice = this.priceToPrecision (symbol, takeProfitPrice);
+                }
+                request['order_configuration'] = {
+                    'stop_limit_stop_limit_gtc': {
+                        'base_size': this.amountToPrecision (symbol, amount),
+                        'limit_price': this.priceToPrecision (symbol, price),
+                        'stop_price': triggerPrice,
+                        'stop_direction': stopDirection,
+                    },
+                };
+            } else {
+                request['order_configuration'] = {
+                    'limit_limit_gtc': {
```

Comment:
Should we support `limit_limit_gtd` and `stop_limit_stop_limit_gtd` order?

Reply:
I don't think it's required since it's only supported on 1 or 2 exchanges, but I added support anyways